metadata-ingestion/src/datahub/cli/specific/user_cli.py

```diff
@@ -55,3 +56,93 @@ def upsert(file: Path, override_editable: bool) -> None:
                     f"Update failed for id {user_config.get('id')}. due to {e}",
                     fg="red",
                 )
+
+
+@user.command(name="add")
+@click.option("--email", required=True, type=str, help="User's email address")
+@click.option(
+    "--display-name", required=True, type=str, help="User's full display name"
+)
+@click.option(
+    "--password",
+    is_flag=True,
+    default=False,
+    help="Prompt for password (hidden input)",
+)
+@click.option(
+    "--role",
+    required=False,
+    type=click.Choice(
+        ["Admin", "Editor", "Reader", "admin", "editor", "reader"], case_sensitive=False
+    ),
+    help="Optional role to assign (Admin, Editor, or Reader)",
+)
+@upgrade.check_upgrade
+def add(email: str, display_name: str, password: bool, role: str) -> None:
+    """Create a native DataHub user with email/password authentication"""
+
+    if not password:
+        click.secho(
+            "Error: --password flag is required to prompt for password input",
+            fg="red",
+        )
+        raise SystemExit(1)
+
+    password_value = click.prompt(
+        "Enter password", hide_input=True, confirmation_prompt=True
+    )
+
+    with get_default_graph(ClientMode.CLI) as graph:
+        user_urn = f"urn:li:corpuser:{email}"
```

Comment:
is using `email` as id a hard assumption?

maybe we should require an `id` field or having a modifier `--email-as-id`